lib/util/index.js

```diff
@@ -104,25 +104,68 @@ function flatMap(list, mapFn) {
   }, []);
 }
 
+/**
+ * Get the browser's user agent, if available.
+ * @returns {?string}
+ */
+function getUserAgent() {
+  return typeof navigator !== 'undefined' && typeof navigator.userAgent === 'string'
+    ? navigator.userAgent
+    : null;
+}
+
 /**
  * Guess the browser.
+ * @param {string} [userAgent=navigator.userAgent]
  * @returns {?string} browser - "chrome", "firefox", "safari", or null
  */
-function guessBrowser() {
-  if (typeof navigator !== 'undefined' && typeof navigator.userAgent === 'string') {
-    if (/Chrome/.test(navigator.userAgent)) {
-      return 'chrome';
-    }
-    if (/Firefox/.test(navigator.userAgent)) {
-      return 'firefox';
-    }
-    if (/Safari/.test(navigator.userAgent)) {
-      return 'safari';
-    }
+function guessBrowser(userAgent) {
```

Comment:
I see that you are using `userAgent` parameter only from unit test. consider changing: 
`function guessBrowser(userAgent)` =>  `function guessBrowser(userAgent = getUserAgent())`

Reply:
@makarandp0 This project is still `es5`, so we can do this once we move to `.ts`.